activerecord/lib/active_record/test_databases.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+require "active_support/testing/parallelization"
+
+module ActiveRecord
+  module TestDatabases
+    ActiveSupport::Testing::Parallelization.after_fork_hook do |i|
+      create_and_migrate(i, spec_name: Rails.env)
+    end
+
+    ActiveSupport::Testing::Parallelization.run_cleanup_hook do |i|
+      drop(i, spec_name: Rails.env)
+    end
+
+    def self.create_and_migrate(i, spec_name:)
+      old, ENV["VERBOSE"] = ENV["VERBOSE"], "false"
+
+      connection_spec = ActiveRecord::Base.configurations[spec_name]
+
+      connection_spec["database"] += "-#{i}"
+      ActiveRecord::Tasks::DatabaseTasks.create(connection_spec)
+      ActiveRecord::Base.establish_connection(connection_spec)
+      if ActiveRecord::Base.connection.migration_context.needs_migration?
+        ActiveRecord::Tasks::DatabaseTasks.migrate
```

Comment:
I think we should always execute (and rely on) `ActiveRecord::Tasks::DatabaseTasks.migrate` to ensure full setup for test database since this creates tables like `schema_migrations`, `ar_internal_metadata` etc.

Reply:
They'll always need a migration because they'll always be brand new. I think however I'm going to drop this implementation and load up the structure/schema instead because it will be faster.